astropy/coordinates/representation.py

```diff
@@ -2029,10 +2068,22 @@ def norm(self):
         """
         return np.abs(self.distance)
 
-    def __neg__(self):
-        self._raise_if_has_differentials('negation')
-        return self.__class__(self.lon + 180. * u.deg, -self.lat, self.distance,
-                              copy=False)
+    def _scale_operation(self, op, *args):
+        # TODO: expand special-casing to UnitSpherical and RadialDifferential.
+        if any(differential.base_representation is not self.__class__
+               for differential in self.differentials.values()):
+            return super()._scale_operation(op, *args)
+
+        lon_op, lat_op, distance_op = _spherical_op_funcs(op, *args)
+
+        result = self.__class__(lon_op(self.lon), lat_op(self.lat),
+                                distance_op(self.distance), copy=False)
+        for key, differential in self.differentials.items():
+            new_comps = (op(getattr(differential, comp)) for op, comp in zip(
```

Comment:
In this case, I have to explicitly apply the operation to the differential, since the change in coordinates is done differently than with a strict scaling.

But perhaps the whole trying to keep the distance positive should go? (or only be done for `__neg__`)

Reply:
Wouldn't that require larger changes to the code? I though distance had to be strictly positive unless specifying a special cosmology or something. I'm probably not grokking what you're suggesting...

Reply:
Hmm, it seems to work fine in the operation. I think just because `Distance` itself does not check immediately. Let's leave that for later clean-up as well...

Reply:
I think I found it. Raw Quantities cannot be negative, but Distances can.